samples/openthread/common/overlay-ot-defaults.conf

```diff
@@ -4,8 +4,8 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
-CONFIG_NCS_SAMPLES_DEFAULTS=y
```

Comment:
Why this change? I guess we should keep logging enabled as default for the other OT samples. Also the lack of these may be the cause of the code size increase for the CoAP client sample:
```
	imply ASSERT_NO_COND_INFO
	imply ASSERT_NO_MSG_INFO
```

Reply:
I removed this because it imply LOG_MINIMAL which causes build warning for NCP/RCP. We want to have deferred logging especially for NCP/RCP to decrease logging impact on the application. I cannot overwrite LOG_MINIMAL in the sample overlay since it's promptless option. What more I belive we should use LOG_MODE_DEFFERED for all samples apart from minimal configuration. It fixes NCP/RCP build where we tried to disable LOG_MINIMAL but it has been set anyway with the build warning.

Reply:
Fair enough.